### PR TITLE
feat: configurable shortcuts with help window

### PR DESCRIPTION
**Pull request by bajor**
## Summary
- All keyboard shortcuts are now configurable via `shortcuts` field in `config.json` — partial overrides merged with defaults
- New `:Raccoon shortcuts` command and `<leader>?` keymap display all active bindings in a floating help window (closes on any keystroke)
- All hardcoded shortcut hints in floating window titles update dynamically to reflect user config

## Changes
- `config.lua`: Added 24 shortcut defaults + `load_shortcuts()` with graceful fallback
- `keymaps.lua`: Refactored from static table to dynamic `build_keymaps(shortcuts)`
- `init.lua`, `diff.lua`, `comments.lua`, `commits.lua`, `review.lua`: All keymaps load from config
- `ui.lua`: Added `show_shortcuts()` floating window + dynamic PR list footer hints
- `plugin/raccoon.lua`: Default config template includes all shortcuts, added `shortcuts` subcommand
- `README.md`: Documented configurable shortcuts with config key names
- `CHANGELOG.md`: Added v0.5 entry

## Test plan
- [x] `make test` passes (63 tests, 0 failures)
- [ ] `:Raccoon config` on fresh install writes all 24 shortcut defaults
- [ ] `:Raccoon shortcuts` opens floating window with grouped shortcut list
- [ ] Floating window closes on any keystroke
- [ ] Override a shortcut in config.json, reopen PR → new binding works
- [ ] All floating window titles reflect configured shortcuts (not hardcoded)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<h2>Greptile Overview</h2>

<h3>Greptile Summary</h3>

This PR adds comprehensive support for configurable keyboard shortcuts to nvim-raccoon, allowing users to customize all 24 keybindings through the `shortcuts` field in `config.json`.

## Key Changes

- **Core Infrastructure**: Added `shortcuts` table to `config.defaults` with 24 keybindings and implemented `load_shortcuts()` function that gracefully falls back to defaults when config is missing or invalid
- **Dynamic Keymaps**: Refactored `keymaps.lua` from static table to dynamic `build_keymaps(shortcuts)` function that constructs keymaps from config at runtime
- **UI Enhancements**: Added `:Raccoon shortcuts` command and `<leader>?` keymap to display all active bindings in a floating help window (closes on any keystroke)
- **Dynamic Hints**: All floating window titles now dynamically reflect user-configured shortcuts instead of hardcoded values (e.g., save/close/resolve labels update to match config)
- **Systematic Integration**: Updated all modules (`init.lua`, `diff.lua`, `comments.lua`, `commits.lua`, `review.lua`) to load shortcuts from config
- **Testing**: Comprehensive test coverage validates defaults, `load_shortcuts()` behavior, partial overrides, and `build_keymaps()` functionality (63 tests pass)
- **Documentation**: README documents all config keys with default values, CHANGELOG has complete v0.5 entry

## Implementation Quality

The implementation follows a clean, consistent pattern across all files:
1. Call `config.load_shortcuts()` to get merged shortcuts (user overrides + defaults)
2. Use shortcut values when building keymaps or window titles
3. Graceful fallback ensures plugin works even without config file

The `load_shortcuts()` function is particularly well-designed - it never fails, always returns valid shortcuts, and doesn't require valid github credentials like `load()` does. This allows the help window to work immediately after installation.

Partial overrides work correctly via `vim.tbl_deep_extend("force", defaults, user_config)`, allowing users to customize only the keys they want while keeping other defaults intact.

<h3>Confidence Score: 5/5</h3>

- This PR is safe to merge with no issues identified
- All changes are well-tested (63 tests pass), follow consistent patterns, maintain backward compatibility through sensible defaults, include comprehensive documentation, and implement a user-requested feature without introducing bugs or breaking existing functionality
- No files require special attention

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| lua/raccoon/config.lua | Adds `shortcuts` field to defaults with 24 configurable keybindings and implements `load_shortcuts()` function that gracefully falls back to defaults. Implementation is clean with proper error handling. |
| lua/raccoon/keymaps.lua | Refactored from static keymaps to dynamic `build_keymaps(shortcuts)` function. All keymaps now load from config via `config.load_shortcuts()` at setup time. Clean separation of concerns. |
| lua/raccoon/ui.lua | Adds `show_shortcuts()` floating window with grouped shortcut display. Dynamic footer hints in PR list. All window titles dynamically reflect configured shortcuts. Well-organized implementation. |
| plugin/raccoon.lua | Adds `shortcuts` subcommand handler and includes all 24 shortcut defaults in config template. Default config generation properly formatted. |
| tests/config_spec.lua | Comprehensive test coverage for shortcuts functionality: validates defaults, tests `load_shortcuts()` with various edge cases, tests partial overrides and merging behavior. All 24 shortcuts validated. |
| tests/keymaps_spec.lua | Tests for `build_keymaps()` function validate dynamic keymap generation, custom shortcut overrides, and proper keymap structure. Good coverage of new functionality. |

</details>

</details>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,23 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 
+## [0.5] - 2026-02-08
```
Comment (bajor) [inline CHANGELOG.md]: test
